test/unit/lib/midnightSpanCorrector.js

```diff
@@ -0,0 +1,171 @@
+const chai = require('chai');
+const chaiMoment = require('chai-moment');
+const Moment = require('moment');
+const OpeningTimes = require('moment-opening-times');
+const midnightSpanCorrector = require('../../../app/lib/midnightSpanCorrector');
+
+chai.use(chaiMoment);
+chaiMoment.setErrorFormat('LLLL');
+
+const expect = chai.expect;
+const aSunday = new Moment('2016-07-24T00:00:00+00:00');
+
+function getAbuttingMidnightDay2359() {
+  return [
+    { opens: '00:00', closes: '20:00' },
+    { opens: '23:00', closes: '23:59' },
+  ];
+}
+
+function getAbuttingMidnightDay0000() {
+  return [
+    { opens: '00:00', closes: '20:00' },
+    { opens: '23:00', closes: '00:00' },
+  ];
+}
+
+function get24HourDay() {
+  return [
+    { opens: '00:00', closes: '23:59' },
+  ];
+}
+
+function getClosedDay() {
+  return [];
+}
+
+function setUpAllWeek(getTimes) {
+  const week = {};
+
+  Moment.weekdays().forEach((d) => {
+    const day = d.toLowerCase();
+    week[day] = getTimes();
+  });
+
+  return week;
+}
+
+function getRegularWorkingWeekAbuttingMidnightAt2359() {
+  const week = setUpAllWeek(getAbuttingMidnightDay2359);
+  week.sunday = getClosedDay();
+  return week;
+}
+
+function getRegularWorkingWeekAbuttingMidnightAt0000() {
+  const week = setUpAllWeek(getAbuttingMidnightDay0000);
+  week.sunday = getClosedDay();
+  return week;
+}
+
+function get24HourWorkingWeekClosedSunday() {
+  const week = setUpAllWeek(get24HourDay);
+  week.sunday = getClosedDay();
+  return week;
+}
+
+function get24HourWorkingWeek() {
+  const week = setUpAllWeek(get24HourDay);
+  return week;
+}
+
+
+function getNewOpeningTimes(openingTimes, timeZone, alterations) {
+  return new OpeningTimes(openingTimes, timeZone, alterations);
+}
+
+function getMoment(day, hours, minutes, timeZone, week) {
+  const dayNumber = Moment
+    .weekdays()
+    .map(d => d.toLowerCase())
+    .indexOf(day);
+  const moment = new Moment(aSunday).tz(timeZone);
+  moment.add(dayNumber, 'days').hours(hours).minutes(minutes);
+  if (week) {
+    moment.add(week, 'week');
+  }
+  return moment;
+}
+
+function momentsShouldBeSame(moment1, moment2) {
+  expect(moment1).to.be.sameMoment(moment2);
+}
+
+describe('Midnight Span Corrector', () => {
```

Comment:
Do we need any tests to show that alternate hours which span midnight are handled?